docs/getting_started/how_to_build_an_api.md

```diff
@@ -0,0 +1,215 @@
+# How to build an API
+
+The purpose of Gosoline is to help with API building. Much of its functionality is API related, and now we will cover the most important parts.
+
+## Package apiserver
+
+An apiserver is, at its bare minimum, a module that runs indefinitely, listens to a port for requests, and provides answers to those requests. 
+
+Package apiserver provides a convenient way to create API server objects:
+
+[embedmd]:# (../../pkg/apiserver/server.go /func New\(/ /\)/)
+```go
+func New(definer Definer)
+```
+
+Where _Definer_ is a method which returns a _*Definitions_:
+
+[embedmd]:# (../../pkg/apiserver/definition.go /type Definer/ /\)\n/)
+```go
+type Definer func(ctx context.Context, config cfg.Config, logger log.Logger) (*Definitions, error)
+```
+
+Lastly, this is how _Definitions_ looks like:
+
+[embedmd]:# (../../pkg/apiserver/definition.go /type Definitions struct/ /\n}/)
+```go
+type Definitions struct {
+	basePath   string
+	middleware []gin.HandlerFunc
+	routes     []Definition
+
+	children []*Definitions
+	parent   *Definitions
+}
+```
+
+
+### Config structs
+
+Some useful configuration structures for api servers:
+
+[structmd]:# (pkg/apiserver/server.go HandlerMetadata Settings TimeoutSettings)
+**HandlerMetadata**
+
+HandlerMetadata stores the Path and Method of this Handler.
+
+| field       | type     | default     | description     |
+| :------------- | :----------: | :----------: | -----------: |
+| Method | string |  | Method is the route method of this Handler. |
+| Path | string |  | Path is the route path ot this handler. |
+
+**Settings**
+
+Settings structure for an API server.
+
+| field       | type     | default     | description     |
+| :------------- | :----------: | :----------: | -----------: |
+| Port | string | 8080 | Port the API listens to. |
+| Mode | string | release | Mode is either debug, release, test. |
+| Compression | CompressionSettings |  | Compression settings. |
+| Timeout | TimeoutSettings |  | Timeout settings. |
+
+**TimeoutSettings**
+
+TimeoutSettings configures IO timeouts.
+
+| field       | type     | default     | description     |
+| :------------- | :----------: | :----------: | -----------: |
+| Read |  | 60s | Read, write and idle timeouts. You need to give at least 1s as timeout.\nRead timeout is the maximum duration for reading the entire request, including the body. |
```

Comment:
Why are there no types here? Also why is there an escaped newline in the description?

Reply:
Then we should add them by hand, or use software which doesn't fall over at the first complication, but that isn't really nice documentation, right?